docs/modules/ROOT/pages/yaml-job-definition.adoc

```diff
@@ -0,0 +1,542 @@
+        type: MAP
+        map: "myMapName"
+----
+
+==== MAP Journal Source
+
+This sources type allows you to work on a entry that is put into defined map. This stage works as a stream.
+
+TIP: This feature requires additional configuration on the map. You should enable _Event Journal_ for your map. link:https://docs.hazelcast.com/hazelcast/latest/data-structures/event-journal#hide-nav[For details.]
+
+[cols="1m,1a,2a,1a"]
+|===
+|Parameter|Required|Description|Default
+
+|type: MAP-JOURNAL
+|Yes
+|Defines the source type of the pipeline as a streamed map.
+|
+
+|map: "myMapName"
+|Yes
+|Name of the map to be used as a data source.
+|
+
+|start-from: 
+|Yes
+|The point where pipeline will start consuming the data from event journal. Options: `OLDEST` or `CURRENT`
+|
+
+|===
+
+[source, yaml]
+----
+pipeline:
+    source:
+        type: MAP-JOURNAL
+        map: "myMapName"
+        start-from: OLDEST
+----
+
+=== Transform Step
+
+In this step, you can shape you data or do computation. The return value will be passed to next step. 
+
+NOTE: Currently, only supported language on transform step is Python. You can transform your data using Python. Also, this step only works with streaming sources.
+
+
+==== Transformation in Python
+
+
+[cols="1m,1a,2a,1a"]
+|===
+|Parameter|Required|Description|Default
+
+|type: PYTHON
+|Yes
+|Defines the transformation type.
+|
+
+|base-image: "hazelcast/python-runtime-base:3.11"
+|Yes
+|The base image to be used to run given Python code. You can customize the image using the base one.
+|
+
+|module: "transformation.transform" 
+|Optional
+|The name of package and transformation function. The format is `package_name.func_name`.
+|transformation.transform
+
+|===
+
+[source,yaml]
+----
+  transform:
+    - type: PYTHON
+      base-image: "hazelcast/python-runtime-base:3.11"
+      module: "transformation.transform"
+----
+
+The Jet Python step expects two functions in your Python code.
+
+[cols="1m,1a,2a,1a"]
+|===
+|Function|Required|Description|Type
+
+|on_setup(config)
+|Optional
+|This function is invoked while runtime is starting. You can register your serializers  to `config` object. `config` object is type of Hazelcast Python client config. link:https://hazelcast.readthedocs.io/en/latest/config.html[For details.] 
+|Hazelcast Python Client Config Object
+
+|transform(data)
+|Yes
+|The function will be invoked when data reaches to the step. If you have a custom type you can prepare and register your serializer in `on_step(config)` function.
+|The `data` argument type depends on previous step in the pipeline.
+
+|===
+
+*Example*
+
+Assume that you have a map journal source, and the source contains `City` object. When a new object put the source map, it will be streamed to the pipeline. In transform step, here, the object type of `data` argument will be a key value pair since source is a map and holds key value pairs. The key value is an integer number in this example, and value is `City` object.
```

Comment:
```suggestion
Assume that you have a map journal source, and the source contains `City` object. When a new object put the source map, it will be streamed to the pipeline. In transform step, the object type of `data` argument will be a key value pair since source is a map and holds key value pairs. The key value is an integer number in this example, and value is `City` object.
```